foldmethod=syntax segfaults

- `nvim --version`: > v0.1.7
- Vim (version: ) behaves differently? Vim doesn't segfault
- Operating system/version: Alpine Linux 3.4
- Terminal name/version: bash 4.3
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
echo -e "void a() {\n}" | nvim -u NORC -
:so $VIMRUNTIME/syntax/c.vim
:set foldmethod=syntax
```

### Actual behaviour
SIGSEGV
### Expected behaviour
A fold is created and editing can continue

Trying to rebuild with more debug output, I noticed these lines fly by during the build. They seem particularly suspicious.

```
 /root/neovim/.deps/usr/include -I/root/neovim/build/src/nvim/auto -I/root/neovim/build/include -I/root/neovim/.deps/usr/include/luajit-2.0  -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -Og -g   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std
=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto  -Wno-static-in-inline -Wno-conversion -o CMakeFiles/nvim.dir/eval.c.o   -c /root/neovim/src/nvim/eval.c                                                                                                                                
/root/neovim/src/nvim/eval.c: In function 'f_foldtext':                
/root/neovim/src/nvim/eval.c:8739:11: warning: implicit declaration of function 'ngettext' [-Wimplicit-function-declaration]                              txt = ngettext("+-%s%3ld line: ", "+-%s%3ld lines: ", count);                                                                                   
           ^                                                                                                                                         /root/neovim/src/nvim/eval.c:8739:9: warning: assignment makes pointer from integer without a cast [-Wint-conversion]                              
     txt = ngettext("+-%s%3ld line: ", "+-%s%3ld lines: ", count);    
         ^                                                                                                                                           /root/neovim/src/nvim/eval.c: In function 'eval6':                                                                                                   
/root/neovim/src/nvim/eval.c:4018:17: warning: 'f2' may be used uninitialized in this function [-Wmaybe-uninitialized]                                                : f1 / f2);                                                                                                                     
                 ^                                                    
/root/neovim/src/nvim/eval.c: At top level:                                                                                                          cc1: warning: unrecognized command line option '-Wno-static-in-inline'
```

--- a/cmake/FindLibIntl.cmake
+++ b/cmake/FindLibIntl.cmake
@@ -46,6 +46,7 @@ check_c_source_compiles("
 
 int main(int argc, char** argv) {
   gettext(\"foo\");
+  ngettext(\"foo\", \"bar\", 1);
   bindtextdomain(\"foo\", \"bar\");
   bind_textdomain_codeset(\"foo\", \"bar\");
   textdomain(\"foo\");

--- a/src/nvim/gettext.h
+++ b/src/nvim/gettext.h
@@ -13,6 +13,7 @@
 #else
 # define _(x) ((char *)(x))
 # define N_(x) x
+# define ngettext(x, xs, n) ((n) == 1 ? (x) : (xs))
 # define bindtextdomain(x, y)  // empty
 # define bind_textdomain_codeset(x, y)  // empty
 # define textdomain(x)  // empty

